lib/kitchen/driver/ec2.rb

```diff
@@ -167,6 +167,15 @@ def default_public_ip_association
         !!config[:subnet_id]
       end
 
+      def default_aws_session_token
+        if config[:aws_secret_access_key] == iam_creds[:aws_secret_access_key] \
+          && config[:aws_access_key_id] == iam_creds[:aws_access_key_id]
+          iam_creds[:aws_session_token]
+        else
```

Comment:
Why is the user required to specify the same access key / secret as returned by the iam_creds query?  If they have to do that, why not just use the provided access key / secret for authentication?

I thought the primary point of the `iam_creds` was they allowed the EC2 node acting as a provisioner to provision nodes using the local credentials.